### PR TITLE
Few fixes and tweaks to tests

### PR DESCRIPTION
**Pull request by renanrodrigo**
## Why is this needed?
This PR solves all of our problems because it addresses specific test failures we saw on Jenkins.
In the end, it adds functionality by letting us choose architectures when running our build scripts.

## Test Steps
- Run CI and see the updated tests pass
- create some other-arch-chroot for ua, and build the package there (:
- Jenkins will be able to build other arches using InstallationSource.LOCAL

## Checklist
 - [ ] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] Changes here need to be documented, and this was done in:

## Does this PR require extra reviews?
 - [ ] Yes
 - [x] No

### CODE REVIEW

File: features/util.py [hunk 1]
```diff
@@ -151,6 +153,7 @@ def repo_state_hash(
 
 def build_debs(
     series: str,
+    architecture: str = get_dpkg_arch(),
```
Comment (orndorffgrant) [inline features/util.py:156]: Currently, passing `architecture=None` like we do down in build.py will overwrite this default value and then we'll get an error trying to create the sbuild command. We can do something like the following to allow the build.py to work:
```suggestion
    architecture: Optional[str] = None,
```
and then
```python
if architecture is None:
    architecture = get_dpkg_arch()
```

**Review by orndorffgrant — COMMENTED** [review]
Small thing, otherwise looks good to me!

File: features/util.py (hunk 1 above)
Comment (renanrodrigo) [reply features/util.py]: Overlooked that - fixed

**Review by lucasmoura — APPROVED** [review]

**Comment by renanrodrigo** [comment]
I'll add a fix to the apt_news test, to make sure we don't expect specific output from apt

**Pull request merged**